pandas/core/arrays/datetimelike.py

```diff
@@ -1359,6 +1359,22 @@ def _addsub_object_array(self, other: np.ndarray, op):
         result = result.reshape(self.shape)
         return result
 
+    def _accumulate(self, name: str, *, skipna: bool = True, **kwargs):
+
+        data = self._ndarray.copy()
+
+        if name in {"cummin", "cummax"}:
+            func = np.minimum.accumulate if name == "cummin" else np.maximum.accumulate
+            result = cast(np.ndarray, nanops.na_accum_func(data, func, skipna=skipna))
```

Comment:
this looks like it might choke on PeriodDtype?

Reply:
Seems to work, could you elaborate what you suspect?

```
ser = Series([pd.Period('2012-1-1', freq='D'), pd.Period('2013-1-1', freq='D')])
ser.cummin()
```

Reply:
i think this goes wrong when there's a NaT present

Reply:
This would restore the previous behavior (but the previous behavior was wrong as well...). Are you ok with addressing this in a follow up?

Reply:
> This would restore the previous behavior

previous as of when?  IIRC this was last changed multiple years ago.

> but the previous behavior was wrong as well

so does this PR get the behavior right for dt64/td64?  If so, the solution for PeriodDtype is similar to median/min/max to do a view to dt64, do the op, then view back.

Reply:
Let's try this one more time, thanks for your patience in explaining this to me.

IIUC this PR should _not_ change any behavior for dt64/dt64tz/td64 dtypes, correct?

Reply:
Yes, correct.